src/Http/Http.Extensions/src/SendFileResponseExtensions.cs

```diff
@@ -124,14 +124,11 @@ private static async Task SendFileAsyncCore(HttpResponse response, string fileNa
 
     private static void CheckRange(long offset, long? count, long fileLength)
     {
-        if (offset < 0 || offset > fileLength)
+        ArgumentOutOfRangeException.ThrowIfLessThan(offset, 0);
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(offset, fileLength);
+        if (count is {} countValue)
         {
-            throw new ArgumentOutOfRangeException(nameof(offset), offset, string.Empty);
-        }
-        if (count.HasValue &&
-            (count.GetValueOrDefault() < 0 || count.GetValueOrDefault() > fileLength - offset))
-        {
-            throw new ArgumentOutOfRangeException(nameof(count), count, string.Empty);
+            ArgumentOutOfRangeException.ThrowIfGreaterThan(countValue, countValue, nameof(count));
```

Comment:
I might be missing something but it appears we've lost the `count > fileLength - offset` check here?

Reply:
@captainsafia,

> I might be missing something but it appears we've lost the `count > fileLength - offset` check here?

Nope! My mistake! Fixed!